src/librustc_target/spec/mod.rs

```diff
@@ -247,11 +247,19 @@ macro_rules! supported_targets {
                         // run-time that the parser works correctly
                         t = Target::from_json(t.to_json())?;
                         debug!("Got builtin target: {:?}", t);
-                        Ok(t)
+                        return Ok(t)
                     },
                 )+
-                _ => Err(format!("Unable to find target: {}", target))
```

Comment:
I believe this would break anyone currently successfully using a non-`unknown` vendor with their own JSON target file. The code before would use their JSON file. The new code would prefer the corresponding builtin target with `unknown` vendor while ignoring their JSON file.

https://github.com/rust-lang/rust/blob/cc4999ea8e6d605e41a1e2e3e846cf633df8a2c2/src/librustc_target/spec/mod.rs#L1101-L1122

I think the fallback to `unknown` would be more appropriate if applies only after an exact match JSON file has not been found.